src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerTest.java

```diff
@@ -1,34 +1,87 @@
 package uk.gov.hmcts.reform.wacaseeventhandler.controllers;
 
+import lombok.extern.slf4j.Slf4j;
 import org.junit.After;
 import org.junit.Test;
 import org.springframework.http.HttpStatus;
 import uk.gov.hmcts.reform.wacaseeventhandler.SpringBootFunctionalBaseTest;
-import uk.gov.hmcts.reform.wacaseeventhandler.domain.EventInformation;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EventInformation;
 
 import java.time.LocalDateTime;
 import java.util.UUID;
 
 import static net.serenitybdd.rest.SerenityRest.given;
 import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 import static uk.gov.hmcts.reform.wacaseeventhandler.CreatorObjectMapper.asJsonString;
 
+@Slf4j
 public class CaseEventHandlerControllerTest extends SpringBootFunctionalBaseTest {
 
-    private final String caseId = UUID.randomUUID().toString();
-    private String taskId;
+    private String taskToTearDown;
 
     @Test
-    public void given_eventInformation_from_ccd_should_respond_with_200() {
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    public void given_initiated_tasks_then_cancel_task() {
+        // Given multiple existing tasks
+
+        // create task1
+        String caseIdForTask1 = UUID.randomUUID().toString();
+        String task1Id = initiateTaskForGivenId(caseIdForTask1);
+
+        // create task2
+        String caseIdForTask2 = UUID.randomUUID().toString();
+        String task2Id = initiateTaskForGivenId(caseIdForTask2);
+
+        // Then cancel the task1
+        String eventToCancelTask = "submitReasonsForAppeal";
+        String previousStateToCancelTask = "awaitingReasonsForAppeal";
+        sendMessage(caseIdForTask1, eventToCancelTask, previousStateToCancelTask);
+
+        // Assert the task1 is deleted
+        assertTaskDoesNotExist(caseIdForTask1);
+        assertTaskDeleteReason(task1Id, "deleted");
+
```

Comment:
Could we also include a check to verify the state of the task is to "Cancelled" (in history table)?

Reply:
That's the way to assert the task is cancelled. I am aware it might not be very clear. In camunda, when the task is cancelled the deleteReason is equal to "delete".